flo-freezer/src/main/java/com/spotify/flo/freezer/PersistingContext.java

```diff
@@ -99,6 +99,7 @@ public static void serialize(Object object, Path file) throws Exception{
     kryo.register(java.lang.invoke.SerializedLambda.class);
     kryo.register(ClosureSerializer.Closure.class, new ClosureSerializer());
     kryo.addDefaultSerializer(java.lang.Throwable.class, new JavaSerializer());
+    kryo.getFieldSerializerConfig().setIgnoreSyntheticFields(false);
```

Comment:
We might not want this to be enabled globally for everything we serialize. We might only want it enabled for 
the TaskBuilder classes and friends in ScalaApi.

Reply:
Here's a post by Nate (creator of Kryo) about the [perils of synthetic fields](https://groups.google.com/d/msg/kryo-users/jgAU18bLyN0/h4jEFSGH0WAJ). However, I we are using the exact same JVM when serializing & deserializing, so I think it might be fine to have it enabled globally.

Reply:
ok, right. we have full control which jvm to use in this case.